pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -314,7 +314,7 @@ func affinityAssistantStatefulSet(name string, pr *v1.PipelineRun, claimTemplate
 					NodeSelector:     tpl.NodeSelector,
 					ImagePullSecrets: tpl.ImagePullSecrets,
 
-					Affinity: getAssistantAffinityMergedWithPodTemplateAffinity(pr),
```

Comment:
What's the reason for this change?

Reply:
The function now needs `aaBehavior` to determine the anti-pod affinity terms

Reply:
I prefer the previous behavior. The new behavior adds more logic to tests, making them more brittle and makes `createOrUpdateAffinityAssistant` and `affinityAssistantStatefulSet` less readable by adding a new argument. (If you do decide to keep the new logic, please add the new argument to the function docstring.)

Reply:
SGTM, I have reverted to the previous behavior!

And also a somewhat related fix: https://github.com/tektoncd/pipeline/pull/6930 🙏